### PR TITLE
Fix the retry / exit logic in maven deploy

### PR DESCRIPTION
**Pull request by hallipr**

### CODE REVIEW

**Review by sima-zhu — APPROVED** [review]
Is it better to sleep some time before retry?

**Review by JimSuplizio — APPROVED** [review]
Thank you for doing this @hallipr!

**Comment by JimSuplizio** [comment]
> Is it better to sleep some time before retry?

I asked the same question however the mvn command, itself, kind of adds some delay. Unfortunately, the maven command can fail even though it's been released. Specifically, we can call the `mvn org.sonatype.plugins:nexus-staging-maven-plugin...` which can start processing, lose connection during that processing which gives us a non-zero exit code, and the test for release can pass. The way things were prior to this PR, the test for release could succeed but would return the non-zero code from the maven command would be returned and this would cause any post-release processing to not happen. The internal pipeline for **java - formrecognizer** (build 20220310.4) is a perfect example of this. The first attempt at release maven was completely down, the second attempt at release it did release however the maven command lost connection before the result was returned. The third attempt failed in spite of the detection logic verifying that the package had been released.

File: eng/scripts/Publish-MavenPackages.ps1 [hunk 1]
```diff
@@ -247,20 +247,17 @@ foreach ($packageDetail in $packageDetails) {
 
       if ($LASTEXITCODE -eq 0) {
         Write-Information "Package $($packageDetail.FullyQualifiedName) deployed"
-        break
+        exit 0
       }
 
-      Write-Information "Release attempt $attemt exited with code $LASTEXITCODE"
+      Write-Information "Release attempt $attempt exited with code $LASTEXITCODE"
```
Comment (benbp) [inline eng/scripts/Publish-MavenPackages.ps1:253]: I don't know if it's an easy add-on here or not, but generally we've been trying to enable powershell strict mode in more places across our scripts.

**Review by benbp — APPROVED** [review]

**Pull request merged**